tests/testthat/_snaps/verb-filter.md

```diff
@@ -28,3 +28,85 @@
       Error in `filter()`:
       ! `.preserve` is not supported on database backends
 
+# filter() after summarise() uses `HAVING`
+
+    Code
+      (out <- lf %>% filter(g == 1))
+    Output
+      <SQL>
+      SELECT `g`, `h`, AVG(`x`) AS `x_mean`
+      FROM `df`
+      GROUP BY `g`, `h`
+      HAVING (`g` = 1.0)
+
+---
+
+    Code
+      (out <- lf %>% filter(x_mean > 1))
+    Output
+      <SQL>
+      SELECT `g`, `h`, AVG(`x`) AS `x_mean`
+      FROM `df`
+      GROUP BY `g`, `h`
+      HAVING (AVG(`x`) > 1.0)
+
+---
+
+    Code
+      (out <- lf %>% filter(g == 1) %>% filter(g == 2))
+    Output
+      <SQL>
+      SELECT `g`, `h`, AVG(`x`) AS `x_mean`
+      FROM `df`
+      GROUP BY `g`, `h`
+      HAVING (`g` = 1.0) AND (`g` = 2.0)
+
+---
+
+    Code
+      (out <- lf %>% filter(g == 1) %>% filter(h == 2))
+    Output
+      <SQL>
+      SELECT `g`, `h`, AVG(`x`) AS `x_mean`
+      FROM `df`
+      GROUP BY `g`, `h`
+      HAVING (`g` = 1.0) AND (`h` = 2.0)
+
+# filter() after mutate() does not use `HAVING`
+
+    Code
+      (out <- lf %>% filter(x_mean > 1))
+    Output
+      <SQL>
+      SELECT *
+      FROM (
+        SELECT `g`, `h`, `x`, AVG(`x`) OVER (PARTITION BY `g`, `h`) AS `x_mean`
+        FROM `df`
+      ) `q01`
+      WHERE (`x_mean` > 1.0)
+
+# filter() using a window function after summarise() does not use `HAVING`
+
+    Code
+      (out <- lf %>% filter(cumsum(x_mean) == 1))
+    Condition
+      Warning:
+      Windowed expression `SUM(`x_mean`)` does not have explicit order.
+      i Please use `arrange()` or `window_order()` to make deterministic.
+    Output
+      <SQL>
+      SELECT `g`, `h`, `x_mean`
+      FROM (
+        SELECT
+          `g`,
+          `h`,
+          `x_mean`,
```

Comment:
Unrelated to this PR, have we considered continuing to use `*` when you only add new columns?

Reply:
No, we haven't. Good point, that would be quite nice.

Reply:
I added PR #881 to tackle this issue.